code/datums/components/construction.dm

```diff
@@ -33,8 +33,9 @@
 
 /datum/component/construction/proc/action(datum/source, obj/item/I, mob/living/user)
 	SIGNAL_HANDLER
-
-	return INVOKE_ASYNC(src, PROC_REF(check_step), I, user)
+	spawn(-1) //Please stop trying to get the return value from procs that can sleep
```

Comment:
I feel like I would rather we have a proc that actually does give a return value because this is only going to serve to confuse everyone who has never seen this construct before (including maintainers) and potentially lead it to be used in places it doesn't belong

Reply:
check_step() has a return value. It sleeps at some point in it's chain, which is why its in a spawn and its return value is treated as valid.

Reply:
I'm talking about `spawn(-1)` itself

Reply:
This is why I would much rather `ImmediateInvokeAsync` give the return value instead

Reply:
imo its more hassle to explain "Using this can return a return value but it also wont under these conditions" than to just educate people on how async works as a whole at this point. At what point are you abstracting so much that it's easier to just explain the core concept than to try to lock it away.